### PR TITLE
Fix 229 relay objects

### PR DESCRIPTION
**Pull request by krishandley**
#229

### CODE REVIEW

**Comment by krishandley** [comment]
Added a dependency so expecting some rolled eyes 🙄 ;) But a simple fix.

My question is why would you ever want to get the relay object in return?

**Comment by krishandley** [comment]
Hmmm actually I'm not sure this fixes the underlying problem. 🤔

**Comment by webark** [comment]
ya.. devDependencies are cheep, so you could even add moment and test against it directly.

But ya, this is going to be a tough one, cause you never really want to return a relay. You just want to return the object with the updated values. Ideally, in my opinion, the relay, and changeset as a whole for that matter, should always return the same object(copy really) you get, and just be there to track the keys that changed. But that requires a change in usage, or architecture..

But maybe i’m wrong..

File: tests/unit/changeset-test.js [hunk 1]
```diff
@@ -90,6 +90,15 @@ test('#get returns change that is a blank value', function(assert) {
   assert.equal(result, '', 'should proxy to change');
 });
 
+test('#get with nested key returns correct value', function(assert) {
+  let date = {};
+  let foo = { date };
+  let dummyModel = EmberObject.create({ foo });
+  let changeset = new Changeset(dummyModel);
+  assert.equal(changeset.get('foo'), foo);
+  assert.equal(changeset.get('foo.date'), date);
```
Comment (webark) [inline tests/unit/changeset-test.js]: I think this should be a deep equals here

**Comment by webark** [comment]
and I think this is the relevant test case that got brought up during the development of nested keys.

https://github.com/poteto/ember-changeset/blob/709be66ae6aae9a0a5c1dc21136da2a51c2f2970/tests/unit/changeset-test.js#L933-L952

File: tests/unit/changeset-test.js (hunk 1 above)
Comment (krishandley) [reply tests/unit/changeset-test.js]: I wanted to check for the exact object, so equal was what I was going for.

**Comment by krishandley** [comment]
Closing. Original fix did work but need to dig deeper to understand the nesting better.

File: tests/unit/changeset-test.js (hunk 1 above)
Comment (webark) [reply tests/unit/changeset-test.js]: ya, but `{} === {}` is false.

Comment (krishandley) [reply tests/unit/changeset-test.js]: That's not what was being tested. foo is already known. `foo === foo` would be true. It's not testing a new empty object against another empty object.

Comment (webark) [reply tests/unit/changeset-test.js]: 🤦‍♂️ yes. Sorry. ... 🐑 ...

**Pull request closed without merging**